rskj-core/src/main/java/co/rsk/validators/BlockUnclesValidationRule.java

```diff
@@ -52,32 +51,30 @@
 
     private static final Logger logger = LoggerFactory.getLogger("blockvalidator");
     private static final PanicProcessor panicProcessor = new PanicProcessor();
+    public static final String INVALIDUNCLE = "invaliduncle";
 
     private BlockStore blockStore;
     private int uncleListLimit;
     private int uncleGenerationLimit;
     private BlockValidationRule validations;
     private BlockParentDependantValidationRule parentValidations;
+    private BlockUnclesHashValidationRule blockValidationRule;
 
     public BlockUnclesValidationRule(BlockStore blockStore, int uncleListLimit, int uncleGenerationLimit, BlockValidationRule validations, BlockParentDependantValidationRule parentValidations) {
         this.blockStore = blockStore;
         this.uncleListLimit = uncleListLimit;
         this.uncleGenerationLimit = uncleGenerationLimit;
         this.validations = validations;
         this.parentValidations = parentValidations;
+        this.blockValidationRule = new BlockUnclesHashValidationRule();
     }
 
     @Override
     public boolean isValid(Block block) {
-        BlockHeader header = block.getHeader();
-        String unclesHash = Hex.toHexString(header.getUnclesHash());
-        String unclesListHash = Hex.toHexString(HashUtil.sha3(header.getUnclesEncoded(block.getUncleList())));
 
-        if (!unclesHash.equals(unclesListHash)) {
-            logger.warn("Block's given Uncle Hash doesn't match: {} != {}", unclesHash, unclesListHash);
-            panicProcessor.panic("invaliduncle", String.format("Block's given Uncle Hash doesn't match: %s != %s", unclesHash, unclesListHash));
+        if (!blockValidationRule.isValid(block))
             return false;
-        }
+
         List<BlockHeader> uncles = block.getUncleList();
         if (CollectionUtils.isNotEmpty(uncles) && !validateUncleList(block.getNumber(), uncles, FamilyUtils.getAncestors(blockStore, block, uncleGenerationLimit), FamilyUtils.getUsedUncles(blockStore, block, uncleGenerationLimit)))
```

Comment:
There's no need to use `CollectionUtils` to check emptyness. You know that `getUncleList` is not null (you should annotate it), so you can call `!uncles.isEmpty()`